src/lib/tcp/src/tests/util/time.rs

```diff
@@ -0,0 +1,404 @@
+//! Time-related types for our unit tests. The types represent simulated times.
+
+// this is only used in doc comments.
+#[allow(unused)]
+use crate::Dependencies;
+
+/// Like [`std::time::Instant`], but the time is arbitrary and is controlled by the
+/// [`Dependencies`].
```

Comment:
I'm not sure what  "controlled by `Dependencies`" means. I can imagine needing this for the ability to create arbitrary values though, which `std::time::Instant` doesn't allow.

Reply:
Yeah this wording is confusing. I removed the "controlled by Dependencies" and added some more comments.